src/theory/strings/infer_proof_cons.cpp

```diff
@@ -652,6 +652,8 @@ void InferProofCons::convert(InferenceId infer,
         }
         else
         {
+          Assert(false)
+              << "Failed to convert length " << lenReq << " " << ps.d_children;
```

Comment:
Intentional? How about `Unreachable`?

Reply:
This should only fail in debug.

The consequence of this failure is that a trusted step will be added to the proof. It is hard to ensure this elaboration always succeeds.